tests/org.jboss.tools.freemarker.ui.bot.test/src/org/jboss/tools/freemarker/ui/bot/test/FreeMarkerCodeAssistTest.java

```diff
@@ -218,20 +220,20 @@ public void after() {
 	private void checkCodeAssist(String expr, String... expected) {
 		log.step("Check if code assist contains expected expressions");		
 		
-		ProjectExplorer pe = new ProjectExplorer();
-		pe.open();
-		new DefaultTreeItem(prj, "ftl", "empty.ftl").doubleClick();				
-		
-		// editor focus bug workaround 
-		pe.open();
-		new DefaultTreeItem(prj, "ftl", "empty.ftl").doubleClick();				
+		PackageExplorer explorer = new PackageExplorer();
+		Project project = explorer.getProject(projectName);
+		project.expand();
```

Comment:
Actually I do, it is only workaround for situation when I get treeItem from PackageExplorer, but in background there is still some thread updating the package/project explorer tree and thus the index of saved tree item is not what was requested. (I can just assume that newly loaded nodes are given used indexies and those of existing nodes are increased). 
For example this exception, It occurs time to time:
org.jboss.reddeer.core.exception.CoreLayerException: There are no items matching matcher tree item text on position <0> matches "import-call-directive.ftl"
	at org.jboss.reddeer.core.lookup.TreeItemLookup.getTreeItems(TreeItemLookup.java:132)
	at org.jboss.reddeer.core.lookup.TreeItemLookup.getTreeItems(TreeItemLookup.java:84)
	at org.jboss.reddeer.core.lookup.TreeItemLookup.getTreeItem(TreeItemLookup.java:63)
	at org.jboss.reddeer.swt.impl.tree.DefaultTreeItem.<init>(DefaultTreeItem.java:47)
	at org.jboss.reddeer.swt.impl.tree.DefaultTreeItem.<init>(DefaultTreeItem.java:37)
	at org.jboss.tools.freemarker.ui.bot.test.FreemarkerDirectiveTest.openFTLFileInEditor(FreemarkerDirectiveTest.java:124)
	at org.jboss.tools.freemarker.ui.bot.test.FreemarkerDirectiveTest.importCallDirectiveOutlineTest(FreemarkerDirectiveTest.java:96)